### PR TITLE
fix(cli): correctly validate "no" response to askDynamoSortKey()

### LINKED ISSUE #1072: storage init: DDB with no LSE includes empty LSE section in the template
If I run storage init for a DDB table:
```
[rear-window-fe] copilot storage init                                                                                             master  ✗ ✭ ✱
Only found one service in workspace, defaulting to: fe
Storage type: DynamoDB
Storage resource name: game
Partition key: game_id
Partition key datatype: String
Sort key? No
✔ Wrote CloudFormation template for DynamoDB Table game at copilot/fe/addons/game.yml
```

The generated CF looks like this:

```yaml
  game:
    Type: AWS::DynamoDB::Table
    DeletionPolicy: Retain
    Properties:
      TableName: !Sub ${App}-${Env}-${Name}-game
      AttributeDefinitions:
        - AttributeName: game_id
          AttributeType: "S"
      BillingMode: PAY_PER_REQUEST
      KeySchema:
        - AttributeName: game_id
          KeyType: HASH
      LocalSecondaryIndexes:
```

The empty `LocalSecondaryIndexes` isn't valid and causes the deployment to fail.

### PR DESCRIPTION
**Pull request by bvtujo**
Ensures that no erroneous "LocalSecondaryIndex" will be printed in CF templates that don't require them (those with no range keys or LSIs specified). 

Fixes #1072 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by bvtujo** [comment]
```  
11   │ Resources:
  12   │   cooltable:
  13   │     Type: AWS::DynamoDB::Table
  14   │     DeletionPolicy: Retain
  15   │     Properties:
  16   │       TableName: !Sub ${App}-${Env}-${Name}-cooltable
  17   │       AttributeDefinitions:
  18   │         - AttributeName: email
  19   │           AttributeType: "S"
  20   │       BillingMode: PAY_PER_REQUEST
  21   │       KeySchema:
  22   │         - AttributeName: email
  23   │           KeyType: HASH
  24   │
```

File: internal/pkg/cli/storage_init.go [hunk 1]
```diff
@@ -212,6 +212,11 @@ func (o *initStorageOpts) Validate() error {
 		}
 	}
 
+	// auto-set noLSI if noSort specified.
+	if o.noSort {
```
Comment (kohidave) [inline internal/pkg/cli/storage_init.go:216]: Hm, I've considered the `validate` method to be more of a read only method.

**Comment by kohidave** [comment]
> Nice work!
> 
> https://github.com/aws/copilot-cli/pull/1076/files#diff-a6941a194c283aee4d8facf12059b601R406-R445
> 
> This seems to be a for loop that will stop once a condition is met. So I think we can get rid of `return nil` outside the for loop block at L445 since it will never reach it.
> 
> Also David, I had a question about
> 
> ```go
> useLSI bool bool
> useSortKey bool
> ```
> 
> It is my understanding that we can't have default values to a struct. Was wondering what `bool bool` does?

Sorry, the double bool was a typo. I just wanted the default value (false) to map to the default choice (no)

File: internal/pkg/cli/storage_init.go (hunk 1 above)
Comment (bvtujo) [reply internal/pkg/cli/storage_init.go]: Fixed in latest.

**Comment by bvtujo** [comment]
> Thanks for the quick fix! I noticed a few areas where I think we can make some small changes to help avoid issues like this in the future!
> 
> https://github.com/aws/copilot-cli/pull/1076/files#diff-a6941a194c283aee4d8facf12059b601R125-R126
> 
> I think maybe the double negative of these bools leads to some confusion. Maybe we can invert them to
> 
> ```go
> useLSI bool bool
> useSortKey bool
> ```
>

This is a little difficult, because it's a tradeoff between defaulting to "no sort key"/"no LSI" for ease of use vs ensuring that folks get the chance to specify every field at initialization that requires replacement on an update. Like, it would be cool to be able to just say 
```
copilot storage init -t DynamoDB -n CoolTable -s fe --partition-key Email:S
```
without having to specify the `--no-sort` and `--no-lsi` flags at the end, but that runs the risk of someone realizing after they ran the command that actually do need at least one sort key, and either deleting the addon and re-running the command, or (yikes) only realizing after they've deployed a table that already has a user. 

> That way the match our default case where the answer is no. I understand that these are flags passed in to folks? If so, maybe we can just add a method that interprets the flags on the struct like `BuildLSI() bool` and `BuildSortKey() bool`

This I can do. I refactored the property parsing logic out of the cli package and into methods in `addon`. 

> 
> https://github.com/aws/copilot-cli/pull/1076/files#diff-a6941a194c283aee4d8facf12059b601L154
> 
> Is this "fs" used?

no, great callout. 
> 
> Would it make sense to break this out into `validateS3` and `validateDDB` or something like that so they're not all lumped together?
> 
> https://github.com/aws/copilot-cli/pull/1076/files#diff-a6941a194c283aee4d8facf12059b601R170-R189

yes!

File: internal/pkg/addon/storage.go [hunk 2]
```diff
@@ -109,3 +113,90 @@ func NewS3(input *S3Props) *S3 {
...
+	p.HasLSI = true
+	lsiConfig, err := newLSI(*p.PartitionKey, lsiSorts)
+	if err != nil {
+		return false, err
+	}
+	p.LSIs = lsiConfig
+	return true, nil
+}
+
+var regexpMatchAttribute = regexp.MustCompile("^(\\S+):([sbnSBN])")
```
Comment (kohidave) [inline internal/pkg/addon/storage.go:173]: Could you move this to the top with the other consts?

File: go.mod [hunk 3]
```diff
@@ -4,7 +4,7 @@ go 1.13
 
 require (
 	github.com/AlecAivazis/survey/v2 v2.0.7
-	github.com/Netflix/go-expect v0.0.0-20190729225929-0e00d9168667 // indirect
+	github.com/aws/amazon-ecs-cli-v2 v0.1.0 // indirect
```
Comment (iamhopaul123) [inline go.mod:7]: Can we remove this?

File: go.sum [hunk 4]
```diff
@@ -39,6 +39,8 @@ github.com/armon/circbuf v0.0.0-20150827004946-bbbad097214e/go.mod h1:3U/XgcO3hC
 github.com/armon/consul-api v0.0.0-20180202201655-eb2c6b5be1b6/go.mod h1:grANhF5doyWs3UAsr3K4I6qtAmlQcZDesFNEHPZAzj8=
 github.com/armon/go-metrics v0.0.0-20180917152333-f0300d1749da/go.mod h1:Q73ZrmVTwzkszR9V5SSuryQ31EELlFMUz1kKyl939pY=
 github.com/armon/go-radix v0.0.0-20180808171621-7fddfc383310/go.mod h1:ufUuZ+zHj4x4TnLV4JWEpy2hxWSpsRywHrMgIH9cCH8=
+github.com/aws/amazon-ecs-cli-v2 v0.1.0 h1:974rhLtDJZ1rCkpdJVCz4U2Fa8HGIh2yDeDAtDr4uBo=
```
Comment (iamhopaul123) [inline go.sum:42]: Also here.

Comment (bvtujo) [reply go.sum:42]: ugh i hate this. sorry this keeps getting committed.

**Pull request merged**